### ISSUE TITLE
DSL is  nice,but  make Newcomer difficult to understand better.

### ISSUE THREAD

**Issue by chrishyc** [raphw/byte-buddy#798]
i  have  read the tutorial carefully. but  i  am  confused with some places.

1.the byte-buddy is woderful.  but  i  donot know  DSL  well.  so  i  have  spent  some time  to search  DLS  articles to understand it.  if  [tutorial](https://bytebuddy.net/#/tutorial)  can  offer  some structure infomations of byte-buddy DSL，i think it.s  better

2.annotions are  nice,but i spent some time to understand Morph. Super ,et.if  there are more  example ,  i will be grateful

**Comment by chrishyc** [comment]
now, i think there are two things important to understand the skeleton of byte-buddy.

1.for  DLS,
(1)method chain of  builder
(2)method chain of loaded/unloaded

2.for the hierarchy of  class , 
many interfaces use  nested abstract class and  nested default implemention class.it is nice to  improve understanding of relative classes.

**Comment by raphw** [comment]
I agree, the documentation lacks a bit but it always falls short. I hope to find some time to rewrite it substantially at some point.
